src/components/features/Home/GoodsRankingSection/GoodsRankingSection.tsx

```diff
@@ -17,23 +19,39 @@ export const GoodsRankingSection = () => {
   });
 
   const [products, setProducts] = useState<product[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [fetchError, setFetchError] = useState<string | null>(null);
 
   useEffect(() => {
     const fetchProducts = async () => {
+      setLoading(true);
+      setFetchError(null);
       try {
         const response = await fetchData('/api/v1/ranking/products', {
           targetType: filterOption.targetType,
           rankType: filterOption.rankType,
         });
         setProducts(response.products);
       } catch (error) {
-        console.error('Error fetching themes:', error);
+        setFetchError((error as Error).message);
+      } finally {
+        setLoading(false);
       }
     };
 
     fetchProducts();
   }, [filterOption]);
 
+  if (loading) {
+    return Loading();
+  }
+  if (fetchError) {
+    return ShowError(fetchError);
+  }
+  if (products.length == 0) {
```

Comment:
```suggestion
  if (products?.length == 0) {
```

혹시 모를 타입 에러를 위해 nullish 체크를 해주면 더 좋겠어요~